packages/panna-sdk/src/core/transaction/transaction.ts

```diff
@@ -0,0 +1,284 @@
+    nonce,
+    extraGas,
+    accessList
+  };
+
+  const prepareParams = {
+    client,
+    chain,
+    ...removeUndefined(optionalFields)
+  };
+
+  return thirdwebPrepareTransaction(prepareParams) as PrepareTransactionResult;
+}
+
+/**
+ * Prepare a contract method call for execution
+ *
+ * This function creates a transaction object for calling a specific method on a smart contract.
+ * It supports both string method signatures and ABI function objects, providing type safety
+ * and parameter validation.
+ *
+ * @param params - Parameters for preparing the contract call
+ * @param params.contract - The contract instance
+ * @param params.method - The method signature or ABI function
+ * @param params.params - The parameters for the method call
+ * @param params.value - The value to send with the transaction (in wei)
+ * @param params.gas - Gas limit for the transaction
+ * @param params.gasPrice - Gas price for legacy transactions
+ * @param params.maxFeePerGas - Maximum fee per gas for EIP-1559 transactions
+ * @param params.maxPriorityFeePerGas - Maximum priority fee per gas for EIP-1559
+ * @param params.nonce - Transaction nonce
+ * @param params.extraGas - Additional gas to add to the estimated gas
+ * @param params.accessList - Access list for EIP-2930 transactions
+ * @returns Prepared contract call transaction
+ *
+ * @example
+ * ```typescript
+ * import { prepareContractCall, getContract, toWei, lisk } from 'panna-sdk';
+ *
+ * // Get a contract instance
+ * const contract = getContract({
+ *   client: pannaClient,
+ *   address: "0x742d35Cc6635C0532925a3b8D42f3C2544a3F97e",
+ *   chain: lisk
+ * });
+ *
+ * // Prepare a contract call with method signature
+ * const transaction = prepareContractCall({
+ *   contract,
+ *   method: "function transfer(address to, uint256 amount)",
+ *   params: ["0x123...", toWei("100")]
+ * });
+ *
+ * // Prepare a payable contract call
+ * const payableCall = prepareContractCall({
+ *   contract,
+ *   method: "function mint(address to)",
+ *   params: ["0x123..."],
+ *   value: toWei("0.1") // 0.1 ETH
+ * });
+ * ```
+ */
+export function prepareContractCall(
+  params: PrepareContractCallParams
+): PrepareContractCallResult {
+  const {
+    contract,
+    method,
+    params: methodParams,
+    value,
+    gas,
+    gasPrice,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonce,
+    extraGas,
+    accessList
+  } = params;
+
+  const contractWithTypedAddress = {
+    ...contract,
+    address: contract.address
+  };
+
+  const optionalFields = {
+    value,
+    gas,
+    gasPrice,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonce,
+    extraGas,
+    accessList
+  };
+
+  const callParams: {
+    contract: {
+      client: PannaClient;
+      address: `0x${string}`;
+      chain: Chain;
+      abi?: Abi;
+    };
+    method: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    params: readonly any[]; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Based on the suggestion above.
```suggestion
    method: `function ${string}`;
    params: readonly any[]; // eslint-disable-line @typescript-eslint/no-explicit-any
```